File: OCTOBER-2025/27-10-2025/p2125.py
class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        arr = []
        for st in bank:
            count = 0
            for s in st:
                if s == '1':
                    count += 1
            if count != 0:           
                arr.append(count)
        
        out = 0
        for i in range(len(arr) - 1):  
            out += arr[i] * arr[i + 1]
        return out

File: OCTOBER-2025/27-10-2025/p2125.java
class Solution {
    public int numberOfBeams(String[] bank) {
        int count = 0, prev =0, curr = 0;
        for(String st : bank){
            curr = 0;
            for(char ch : st.toCharArray()){
                if(ch == '1'){
                    curr++;
                }
            }
            count+=(curr * prev);
            if(curr>0){
                prev = curr;
            }
        }
        return count;
    }
}